lib/finch/http1/pool.ex

```diff
@@ -16,36 +17,63 @@ defmodule Finch.HTTP1.Pool do
     }
   end
 
-  def start_link({shp, registry_name, pool_size, conn_opts, pool_max_idle_time}) do
+  def start_link({shp, registry_name, pool_size, conn_opts, pool_max_idle_time, pool_metrics}) do
     NimblePool.start_link(
-      worker: {__MODULE__, {registry_name, shp, conn_opts}},
+      worker: {__MODULE__, {registry_name, shp, conn_opts, pool_size, pool_metrics}},
       pool_size: pool_size,
       lazy: true,
       worker_idle_timeout: pool_idle_timeout(pool_max_idle_time)
     )
   end
 
   @impl Finch.Pool
-  def request(pool, req, acc, fun, opts) do
+  def request(pool, req, acc, fun, name, opts) do
     pool_timeout = Keyword.get(opts, :pool_timeout, 5_000)
     receive_timeout = Keyword.get(opts, :receive_timeout, 15_000)
-
     metadata = %{request: req, pool: pool}
```

Comment:
Since we're already passing the Finch name in, could we add it to the metadata so it's available in telemetry? It would be really useful in monitoring wait time per Finch instance?

Reply:
Yes exactly @aselder I was thinking these changes enable #199 . I think that should be implemented in a new PR once this is complete.

Reply:
I think this is a good idea indeed. But just to be clear, I've only added the finch_name on the Finch.Pool request callback because of averages and maxes metrics.

If we gonna move on only with available and in use connections this is not necessary.

Should I keep it here unused anyway in order to enable this change? What you think?

Reply:
Considering the modifications on https://github.com/sneako/finch/pull/252 I think would be better to remove these changes from this PR in order to avoid conflicts.